appinventor/buildserver/src/com/google/appinventor/buildserver/Compiler.java

```diff
@@ -1016,7 +1045,24 @@ private boolean runDx(File classesDir, String dexedClasses) {
     synchronized (SYNC_KAWA_OR_DX) {
       setProgress(50);
       dxSuccess = dexTask.execute(inputList);
-      setProgress(75);
+      if (dxSuccess && (class2List.size() > 0)) {
+        setProgress(60);
+        dexTask.setOutput(dexedClassesDir + File.separator + "classes2.dex");
+        inputList = new ArrayList<File>();
+        dxSuccess = dexTask.execute(class2List);
+        setProgress(75);
+        hasSecondDex = true;
+      } else if (!dxSuccess) {  // The initial dx blew out, try more conservative
+        LOG.info("DX execution failed, trying with fewer libraries.");
```

Comment:
I don't get what "fewer libraries" means here.   I guess it's related to my comment above.